management/src/main/java/nearsoft/skytouch/management/controller/ProductController.java

```diff
@@ -0,0 +1,49 @@
+package nearsoft.skytouch.management.controller;
+
+import nearsoft.skytouch.common.model.Product;
+import nearsoft.skytouch.management.service.ProductService;
+import org.springframework.stereotype.Controller;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import java.util.List;
+import java.util.Map;
+
+
+@Controller
+class ProductController {
+
+    private ProductService productService;
+
+    public ProductController(ProductService productService) {
```

Comment:
How are you giving a value to productService?

Reply:
ProductServiceImpl is annotated as a Service so it is injected here

Reply:
Annotating a class as a service turns that class into a bean in the application context, so that springboot can manage it, to make us of springboot's dependency injection you should be autowiring that service on this class.

Reply:
I think it is no necessary use the Autowire annotation in this case, as you can see here https://docs.spring.io/spring-boot/docs/current/reference/html/using-boot-spring-beans-and-dependency-injection.html
if there are only  one constructor that annotation it's not needed

Reply:
I would still use the annotation, just for clarity, however, that is up to you then. Thank you.